internal/oracle/entity_test.go

```diff
@@ -0,0 +1,353 @@
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			err: nil,
+		},
+		{
+			name: "Management type",
+			sub: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 2,
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			full: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				CountryCode:  "AUS",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 1,
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			err: errors.New("Manager 0"),
+		},
+		{
+			name: "Extra Management",
+			sub: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 2,
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			full: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				CountryCode:  "AUS",
+			},
+			err: errors.New("Manager 0"),
+		},
+		{
+			name: "Second Management",
+			sub: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 2,
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			full: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				CountryCode:  "AUS",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 5,
+						Name: "John",
+					},
+					&actions.ManagerField{
+						Type: 2,
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			err: nil,
+		},
+		{
+			name: "Out of order Management",
+			sub: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 2,
+						Name: "Satoshi Nakamoto",
+					},
+					&actions.ManagerField{
+						Type: 5,
+						Name: "John",
+					},
+				},
+			},
+			full: &actions.EntityField{
+				Name:         "Tokenized",
+				EmailAddress: "satoshi@tokenized.com",
+				CountryCode:  "AUS",
+				Management: []*actions.ManagerField{
+					&actions.ManagerField{
+						Type: 5,
+						Name: "John",
+					},
+					&actions.ManagerField{
+						Type: 2,
+						Name: "Satoshi Nakamoto",
+					},
+				},
+			},
+			err: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.sub.Validate(); err != nil {
+				t.Errorf("Sub invalid : %s", err)
+				return
+			}
+			if err := tt.full.Validate(); err != nil {
+				t.Errorf("Full invalid : %s", err)
+				return
+			}
+
+			err := VerifyEntityIsSubset(tt.sub, tt.full)
+
+			if tt.err == nil {
+				if err != nil {
+					t.Errorf("Entity didn't verify : %s", err)
+				}
+			} else {
```

Comment:
No need for else. 

Just t.Fatal above to stop this test.

Reply:
This else is needed to check for the cases that are supposed to cause an error. I will add log statements to make that more clear.